src/java.base/share/classes/java/io/Reader.java

```diff
@@ -391,6 +397,71 @@ public int read(char[] cbuf) throws IOException {
      */
     public abstract int read(char[] cbuf, int off, int len) throws IOException;
 
+    /**
+     * Reads all remaining characters as lines of text. A line is considered to
+     * be terminated by any one of a line feed ('\n'), a carriage return ('\r'),
+     * a carriage return followed immediately by a line feed, or by reaching the
+     * end-of-file (EOF).
+     *
+     * <p> This method works as if invoking it were equivalent to evaluating
+     * the expression:
+     * <blockquote>{@linkplain #readAllChars()}.lines().toList()</blockquote>
```

Comment:
I think this is actually too specific, as it could imply the returned List is the exact implementation that is returned by Stream.toList(). It is in this implementation, but I could easily imagine another implementation that constructs the list a different way. It might be better to be descriptive here and say that the entire input is read and is split into lines (where "line" is defined as above) which are returned in an unmodifiable list. And then have "unmodifiable list" be a link to the Unmodifiable Lists section, something like "java/util/List.html#unmodifiable".

Reply:
The word 'equivalent' mitigates the exactness of the code to describe the expected result.

Reply:
I think it's still ambiguous. There was a back-and-forth a while ago -- I think @bplb was involved -- where there was similar wording about something being "equivalent to" or "as if" foo() was called, and the person was insistent that this meant that foo() should actually be called (and it was a bug that foo() was not called).

A possible rewording that might tip the balance is to change "works as if" (which implies implementation) to "the result is as if" (which emphasizes the result and less how it was arrived at). But I'd still recommend avoiding using code as a specification, especially if the code involves a publicly overridable method.

Reply:
> I think it's still ambiguous. [...]

Without addressing the specific comments, I would simply like to point out that "as if" and "equivalent" is common phrasing in this area, e.g.,

- [File.toPath](https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/io/File.html#toPath())
- [Files.newBufferedReader](https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/nio/file/Files.html#newBufferedReader(java.nio.file.Path))
- etc.